src/WebPush.php

```diff
@@ -121,93 +121,53 @@ public function sendNotification(Subscription $subscription, ?string $payload =
 
         $this->notifications[] = new Notification($subscription, $payload, $options, $auth);
 
-        if ($flush) {
-            $res = $this->flush();
-
-            // if there has been a problem with at least one notification
-            if (is_array($res)) {
-                // if there was only one notification, return the information directly
-                if (count($res) === 1) {
-                    return $res[0];
-                }
-
-                return $res;
-            }
-
-            return true;
-        }
-
-        return true;
+	    return false !== $flush ? $this->flush() : true;
     }
 
-    /**
-     * Flush notifications. Triggers the requests.
-     *
-     * @param null|int $batchSize Defaults the value defined in defaultOptions during instantiation (which defaults to 1000).
-     *
-     * @return array|bool If there are no errors, return true.
-     *                    If there were no notifications in the queue, return false.
-     * Else return an array of information for each notification sent (success, statusCode, headers, content)
-     *
-     * @throws \ErrorException
-     */
-    public function flush(?int $batchSize = null)
+	/**
+	 * Flush notifications. Triggers the requests.
+	 *
+	 * @param null|int $batchSize Defaults the value defined in defaultOptions during instantiation (which defaults to 1000).
+	 *
+	 * @return \Generator
+	 * @throws \ErrorException
+	 */
+    public function flush(?int $batchSize = null) : \Generator
```

Comment:
`\iterable` is just a pseudo type and generators implement that type.
See ex. 4 of the link I gave.

Having `\iterable` instead of `\Generator` in the method signature won't change anything regarding the memory saving or the behaviour of the `yield` used in the method.
You will iterate the same way and `foreach ($res as $result) {` in your example will still be valid.

From my POV, it is preferable to indicate pseudo-types as return types instead of a concrete/specific type. As I mentioned, it won't change anything, but it will be easier to mock or override the class with the later.

Reply:
> but it will be easier to mock or override the class with the later.
Fair enough. Will try to change in nearest future.

Reply:
Change applied.

Reply:
Excellent!
Thank you.